client/components/signup-form/index.jsx

```diff
@@ -67,39 +63,48 @@ export default React.createClass( {
 		submitButtonText: PropTypes.string.isRequired,
 		submitting: PropTypes.bool,
 		suggestedUsername: PropTypes.string.isRequired,
-	},
-
-	getDefaultProps() {
-		return {
-			isSocialSignupEnabled: false,
-		};
-	},
-
-	getInitialState() {
-		return {
-			notice: null,
-			submitting: false,
-			form: null,
-			signedUp: false,
-			validationInitialized: false
-		};
-	},
+		translate: PropTypes.func.isRequired
+	};
+
+	static defaultProps = {
+		isSocialSignupEnabled: false,
+	};
+
+	state = {
+		notice: null,
+		submitting: false,
+		form: null,
+		signedUp: false,
+		validationInitialized: false
+	};
+
+	constructor( props ) {
+		super( props );
+
+		this.handleBlur = this.handleBlur.bind( this );
```

Comment:
Since you changed all those occurrences. You can use arrow functions instead of binding to `this`:
```js
handleBlur = () {
...
};
```